sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/TracerProvider.java

```diff
@@ -121,18 +129,28 @@ public Context startSpan(String methodName, String databaseId, String endpoint,
     public <T> Mono<CosmosItemResponse<T>> traceEnabledCosmosItemResponsePublisher(Mono<CosmosItemResponse<T>> resultPublisher,
                                                                                    Context context,
                                                                                    String spanName,
+                                                                                   String containerId,
                                                                                    String databaseId,
-                                                                                   String endpoint) {
-        return traceEnabledPublisher(resultPublisher, context, spanName, databaseId, endpoint,
+                                                                                   CosmosAsyncClient client,
+                                                                                   ConsistencyLevel consistencyLevel,
+                                                                                   OperationType operationType,
+                                                                                   ResourceType resourceType) {
+
+        return publisherWithClientTelemetry(resultPublisher, context, spanName, containerId, databaseId,
+            BridgeInternal.getServiceEndpoint(client),
+            client,
+            consistencyLevel,
+            operationType,
+            resourceType,
             CosmosItemResponse::getStatusCode);
     }
 
-    public <T> Mono<T> traceEnabledPublisher(Mono<T> resultPublisher,
-                                             Context context,
-                                             String spanName,
-                                             String databaseId,
-                                             String endpoint,
-                                             Function<T, Integer> statusCodeFunc) {
+    private <T> Mono<T> traceEnabledPublisher(Mono<T> resultPublisher,
```

Comment:
Why are we changing the scope of this API ?

Reply:
This must be left public my mistake , not getting called from outside